instrumentation/apache-httpclient/apache-httpclient-4.3/library/src/main/java/io/opentelemetry/instrumentation/apachehttpclient/v4_3/ApacheHttpClientTracingBuilder.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.apachehttpclient.v4_3;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.instrumentation.api.instrumenter.AttributesExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+import io.opentelemetry.instrumentation.api.instrumenter.SpanKindExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.SpanNameExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.SpanStatusExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.http.HttpAttributesExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.http.HttpSpanNameExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.http.HttpSpanStatusExtractor;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.http.HttpResponse;
+
+/** A builder for {@link ApacheHttpClientTracing}. */
+public final class ApacheHttpClientTracingBuilder {
+
+  private static final String INSTRUMENTATION_NAME = "io.opentelemetry.apache-httpclient-4.3";
+
+  private final OpenTelemetry openTelemetry;
+
+  private final List<AttributesExtractor<? super ApacheHttpClientRequest, ? super HttpResponse>>
+      additionalExtractors = new ArrayList<>();
+
+  ApacheHttpClientTracingBuilder(OpenTelemetry openTelemetry) {
+    this.openTelemetry = openTelemetry;
+  }
+
+  /**
+   * Adds an additional {@link AttributesExtractor} to invoke to set attributes to instrumented
+   * items. The {@link AttributesExtractor} will be executed after all default extractors.
+   */
+  public ApacheHttpClientTracingBuilder addAttributeExtractor(
+      AttributesExtractor<? super ApacheHttpClientRequest, ? super HttpResponse>
```

Comment:
Should `ApacheHttpClientRequest` be public? We're exposing a possibility to add a custom attributes extractor, but you can't really use request in it.

Reply:
@trask This is an interesting case for wanting to go back to the HttpUriRequest interface perhaps :-)

For now I just made public with a single method returning the delegate.

Reply:
i'm not sure wrangling into `HttpUriRequest` is ideal for custom attribute extractors either, I think ideally they would want access to `HttpHost` + `HttpRequest` in the case where an `HttpHost` + `HttpRequest` overload is used